### PR TITLE
[Merged by Bors] - feat: case-specific equivalences for `IsROrC`

### PR DESCRIPTION
**Pull request by j-loreaux**
This adds ring isomorphisms and linear isometry equivalences between `𝕜` satisfying `IsROrC 𝕜` and `ℝ` or `ℂ` depending on whether `I = 0` or `im I = 1`. Of course, the design of `IsROrC` is meant to eliminate the need for such things most of the time, but these are helpful in the rare case one actually does need to split into the two cases in the middle of a proof.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/IsROrC/Basic.lean [hunk 1]
```diff
@@ -1139,4 +1143,36 @@ theorem continuous_normSq : Continuous (normSq : K → ℝ) :=
...
+section CaseSpecific
+
+lemma im_eq_zero (h : I = (0 : K)) (z : K) : im z = 0 := by
+  rw [← re_add_im z, h]
+  simp
+
+/-- The natural isomorphism between `𝕜` satisfying `IsROrC 𝕜` and `ℝ` when `IsROrC.I = 0`. -/
+@[simps]
+def realRingEquiv (h : I = (0 : K)) :
+    K ≃+* ℝ where
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/IsROrC/Basic.lean:1160]: Any reason not to do this?
```suggestion
def realRingEquiv (h : I = (0 : K)) : K ≃+* ℝ where
```

**Review by Ruben-VandeVelde — APPROVED** [review]

File: Mathlib/Analysis/Complex/Basic.lean [hunk 2]
```diff
@@ -451,6 +454,34 @@ lemma exists_norm_eq_mul_self (z : ℂ) : ∃ c, ‖c‖ = 1 ∧ ‖z‖ = c * z
 lemma exists_norm_mul_eq_self (z : ℂ) : ∃ c, ‖c‖ = 1 ∧ c * ‖z‖ = z :=
   IsROrC.exists_norm_mul_eq_self _
 
+/-- The natural isomorphism between `𝕜` satisfying `IsROrC 𝕜` and `ℂ` when
+`IsROrC.im IsROrC.I = `. -/
```
Comment (ADedecker) [inline Mathlib/Analysis/Complex/Basic.lean:458]:
```suggestion
`IsROrC.im IsROrC.I = 1`. -/
```

**Review by ADedecker — COMMENTED** [review]
As I commented [here](https://github.com/leanprover-community/mathlib4/pull/9788#issuecomment-1894477138), I think you don't actually need this for #9788, but this is nice to have anyways.

Thanks!

bors d+

**Comment by ADedecker** [comment]
bors r-

I messed up

bors d+

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**